task/gc_hook.go

```
@@ -0,0 +1,26 @@
+// durably, install a Store with SetStore instead: Store.SaveRun is handed the
+// same snapshot slice from a background goroutine with no lock held, so it may
+// block on IO, and it fires the moment a run finishes rather than at eviction
+// runRetention later. Reach for OnBeforeGC when the work has to happen at
+// eviction specifically — releasing resources keyed by run id, dropping caches,
+// cleaning up scratch directories.
+//
+// The two are independent: when both are installed, both are called.
+var OnBeforeGC func(groupID string, snapshots []TaskSnapshot)
+
+// notifyBeforeGC delivers an evicted group to the hook, if one is installed.
+func notifyBeforeGC(groupID string, snapshots []TaskSnapshot) {
```

golangci-lint (2.13.2)

[error] func notifyBeforeGC is unused

(unused)